crates/rpc/rpc-types/src/eth/block.rs

```diff
@@ -33,6 +45,80 @@ pub struct Block {
     pub base_fee_per_gas: Option<U256>,
 }
 
+impl Block {
+    /// Create a new block response from a [primitive block](reth_primitives::Block), using the
+    /// total difficulty to populate its field in the rpc response.
+    pub fn from_block_full(
+        block: PrimitiveBlock,
+        total_difficulty: U256,
+    ) -> Result<Self, BlockError> {
+        let block_hash = block.header.hash_slow();
+        let header_length = block.header.length();
+        let block_length = block.length();
+        let uncles = block.ommers.into_iter().map(|h| h.hash_slow()).collect();
+
+        let RethHeader {
+            parent_hash,
+            ommers_hash,
+            beneficiary,
+            state_root,
+            transactions_root,
+            receipts_root,
+            logs_bloom,
+            difficulty,
+            number,
+            gas_limit,
+            gas_used,
+            timestamp,
+            mix_hash,
+            nonce,
+            base_fee_per_gas,
+            extra_data,
+        } = block.header;
+
+        let header = Header {
+            hash: Some(block_hash),
+            parent_hash,
+            uncles_hash: ommers_hash,
+            author: beneficiary,
+            miner: beneficiary,
+            state_root,
+            transactions_root,
+            receipts_root,
+            number: Some(U256::from(number)),
+            gas_used: U256::from(gas_used),
+            gas_limit: U256::from(gas_limit),
+            extra_data,
+            logs_bloom,
+            timestamp: U256::from(timestamp),
+            difficulty,
+            mix_hash,
+            nonce: Some(nonce.to_be_bytes().into()),
+            size: Some(U256::from(header_length)),
+        };
```

Comment:
IF we really need both types, `impl From`

Reply:
in this case, we can't implement `From` directly because we would need the total difficulty, which is currently only stored the db, not in the `Block` type

Reply:
I guess this is cheap enough and is fine